server/app.js

```diff
@@ -25,8 +27,30 @@ const io = socketio(server, {
   },
 });
 
+io.use(async (socket, next) => {
+  if (socket.handshake.headers.cookie) {
+    token = cookie.parse(socket.handshake.headers.cookie).token;
+    if (token) {
+      const authorized = await jwt.verify(token, process.env.JWT_SECRET);
```

Comment:
Seems [`jwt.verify` is executed synchronously](https://www.npmjs.com/package/jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback) unless it's provided a callback, so we should remove the `await` keyword here 👍🏼 

Another note, `jwt.verify` will throw if the token is not valid so we will need to wrap it in a `try...catch` block. You can also simplify your conditional statements based on this - if it doesn't throw, then you can assume the token is valid and won't have to check for `authorized.id`👍🏼 

**Aside**: [This stackoverflow answer](https://stackoverflow.com/a/53113299) covers more about the impact of using await on synchronous functions.

Reply:
Changes made. Thanks for the resources.